lib/logflare/backends/adaptor/datadog_adaptor.ex

```diff
@@ -0,0 +1,190 @@
+defmodule Logflare.Backends.Adaptor.DatadogAdaptor do
+  @moduledoc """
+  Wrapper module for `Logflare.Backends.Adaptor.WebhookAdaptor` to provide API
+  for DataDog logs ingestion endpoint.
+  """
+
+  defmodule Client do
+    use GenServer
+
+    require Logger
+
+    # - `queue` - list of elements in queue
+    # - `size` - amount of elements in queue (cached to not use `length/1` all
+    #   the time)
+    # - `threshold` - maximal amount of items in queue before sending
+    # - `timeout` - time to wait before sending all messages in queue
+    # - `timer` - reference to timer used
+    defstruct queue: [],
+              size: 0,
+              threshold: 100,
+              timeout: 1000,
+              timer: nil,
+              url: nil,
+              method: :post,
+              headers: []
+
+    def start_link(opts), do: GenServer.start_link(__MODULE__, Map.new(opts))
+
+    def ingest(_pid, []), do: :ok
+    def ingest(pid, logs), do: GenServer.cast(pid, {:events, logs})
```

Comment:
This would get called for even small requests or singular event requests, as `Backends.ingest_logs/2` doesn't handle batching.
would need to handle batching at adaptor level, which is why i've mostly been using Broadway to auto-handle them. This won't survive at prod scale, which is the issues facing the current WebhookAdaptor and PostgresAdaptor.

Reply:
Yesm that would get called for singular events, that is why that module *is* implementing batching in a way that it will wait until there is given amount of events in queue **or** timeout was reached. The goal of the timeout is to not batch messages indefinitely if there will be not enough of them, but rather send messages in more or less uniform periods even if there is less than N messages.

Reply:
this is the same as the `:batch_timeout` option for broadway

Reply:
Oh, TIL. I can change it to Broadway to make everything more uniform